services/app/src/hooks/useGameId.js

```diff
@@ -1,11 +1,24 @@
 import { useCallback } from 'react';
+import { nanoid } from 'nanoid';
+
+const ROOM_ID_LENGTH = 8;
 
 export const useGameId = () => {
```

Comment:
What do you think if we were to refactor this to use a more react-like approach with state.

e.g.
```
export const useGameId = () => {
  const generateGameId = () => nanoid(ROOM_ID_LENGTH);
  const [gameId, _setGameId] = useState(localStorage.getItem('lastGameId') ?? generateGameId());

  const setGameId = useCallback((id) => {
    window.localStorage.setItem('lastGameId', id);
    _setGameId(id);
  }, []);

  return { gameId, setGameId };
};
```

We would not need to call the `getGameId` as a function to obtain the id, but rather use react's state which will allow us to rely on the `gameId` variable to rerun `useEffects` that should be ran when `gameId` changes for any reason.

Reply:
Yes! I like this approach, will do that.